src/core/isIterable.js

```diff
@@ -0,0 +1,10 @@
+/** @license ISC License (c) copyright 2018 original and current authors */
+/** @author Dale Francis (dalefrancis88) */
+
+const isFunction = require('./isFunction')
+
+function isIterable(iterable) {
+  return iterable !== null && iterable !== undefined && isFunction(iterable[Symbol.iterator])
```

Comment:
with `NaN` also not being iterable, could we use the `isNil` predicate here?

Reply:
cool, good idea, will do it